### PR TITLE
Bot : currencies/family filter, disable & mooncake/silicium non reg removed

### PR DESCRIPTION
**Pull request by hedi-edelbloute**
### 📝 Description

Various update to the bot and workflows so :

- we can **filter** by a list of currencies/families
- we can **disable** a list of currencies/families
- we can see in the action how we run the bot : **daily / weekly basis**
- **disabled job starts on push** because we use workflow dispatch
- non reg on **mooncake, silicium disabled** as we already run those tests on other seeds

**Disabled on non reg** : Bitcoin, Ethereum as it costs too much and we test on testnets

**Weekly basis**: qtum,decred,cardano,axelar,cosmos,secret_network,avalanche_c_chain,bsc,filecoin,tron,cronos,fantom,boba,telos_evm,polygon_zk_evm,polkadot

**Daily basis** : the rest

I ignored coins that don't return addresses (evmos, flare...) as it is related to a problem

### ❓ Context

- **Impacted projects**: `LLC` 
- **Linked resource(s)**: `https://ledgerhq.atlassian.net/browse/LIVE-8283` 

### ✅ Checklist

- [x] **Test coverage** 
- [x] **Atomic delivery** 
- [x] **No breaking changes** 

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

### CODE REVIEW

File: .github/workflows/bot-nitrogen.yml [hunk 1]
```diff
@@ -1,20 +1,22 @@
...
-run-name: Test with 🤖 Nitrogen by @${{github.actor}} with ${{inputs.family}}
+run-name: Test with 🤖 Nitrogen by @${{github.actor}} family=${{inputs.families}} currency=${{inputs.currencies}}
 on:
   workflow_dispatch:
     inputs:
-      family:
-        description: coin family to filter with (if any)
+      ref:
+        description: ref branch to run on
+        required: true
```
Comment (valpinkman) [inline .github/workflows/bot-nitrogen.yml:8]: not required

File: .github/workflows/bot-nitrogen.yml [hunk 2]
```diff
@@ -1,20 +1,22 @@
...
+        required: false
+      currencies:
+        description: currency ids to filter, separated with commas (,)
         required: false
       speculos_websocket:
         description: check to use speculos with Websocket
         type: boolean
         required: false
         default: false
-  push:
```
Comment (valpinkman) [inline .github/workflows/bot-nitrogen.yml:14]: why did you remove on: push ?

**Review by valpinkman — CHANGES_REQUESTED** [review]
missing updates on the composite to make it work

File: .github/workflows/bot-nitrogen.yml (hunk 2 above)
Comment (hedi-edelbloute) [reply .github/workflows/bot-nitrogen.yml:14]: We always use workflow dispatch

File: .github/workflows/bot-nitrogen.yml (hunk 1 above)
Comment (hedi-edelbloute) [reply .github/workflows/bot-nitrogen.yml]: Used only for testing

File: .github/workflows/bot-nonreg-nitrogen.yml [hunk 3]
```diff
@@ -34,3 +33,4 @@ jobs:
           SLACK_ICON_EMOJI: ":bot-seed7:"
           SEED: ${{ secrets.SEED7 }}
           BOT_ENVIRONMENT: production
+          BOT_DISABLED_CURRENCIES: bitcoin,ethereum,qtum,decred,cardano,axelar,cosmos,secret_network,avalanche_c_chain,bsc,filecoin,tron,cronos,fantom,boba,telos_evm,polygon_zk_evm,polkadot
```
Comment (lvndry) [inline .github/workflows/bot-nonreg-nitrogen.yml:36]: Why is this needed ?

File: .github/workflows/bot-nitrogen.yml (hunk 2 above)
Comment (chabroA) [reply .github/workflows/bot-nitrogen.yml:14]: @valpinkman what is the purpose of this `push` configuration?

At first I thought:

> It looks like this (and similar things on other workflows) are here for a reason and used for easier support and maintenance.

But since these workflows are triggerd manually through workflow dispatch, it doesn't seem needed because we can select the branch from which to execute the workflow on github if we want to test our updates before merging them.

If this is still needed, could you provide details so it can be included in a comment? Thanks

File: .github/workflows/bot-oxygen.yml [hunk 4]
```diff
@@ -1,15 +1,19 @@
 name: "[Bot] Testing with 'Oxygen'"
-run-name: Test with 🤖 Oxygen by @${{github.actor}} with ${{inputs.family}}
+run-name: Test with 🤖 Oxygen by @${{github.actor}} ${{inputs.families &&  format('family={0}',inputs.families) || ''}} ${{inputs.currencies && format('currency={0}',inputs.currencies) || ''}}
 on:
   workflow_dispatch:
     inputs:
-      family:
-        description: coin family to filter with (if any)
+      families:
+        description: coin family filter, separated with commas (,)
```
Comment (chabroA) [inline .github/workflows/bot-oxygen.yml:7]: Similar comment as made on "mooncake"

File: .github/workflows/bot-oxygen.yml [hunk 5]
```diff
@@ -1,15 +1,19 @@
...
         required: false
-  push:
-    branches:
-      - support/bot-oxygen
-      - support/bot-oxygen-*
+      currencies:
+        description: currency ids to filter, separated with commas (,)
```
Comment (chabroA) [inline .github/workflows/bot-oxygen.yml:10]: Similar comment as made on "mooncake"

File: .github/workflows/bot-phosphore.yml [hunk 6]
```diff
@@ -1,15 +1,19 @@
 name: "[Bot] Testing with 'Phosphore'"
-run-name: Test with 🤖 Phosphore by @${{github.actor}} with ${{inputs.family}}
+run-name: Test with 🤖 Phosphore by @${{github.actor}} ${{inputs.families &&  format('family={0}',inputs.families) || ''}} ${{inputs.currencies && format('currency={0}',inputs.currencies) || ''}}
 on:
   workflow_dispatch:
     inputs:
-      family:
-        description: coin family to filter with (if any)
+      families:
+        description: coin family filter, separated with commas (,)
```
Comment (chabroA) [inline .github/workflows/bot-phosphore.yml:7]: Similar comment as made on "mooncake"

File: .github/workflows/bot-phosphore.yml [hunk 7]
```diff
@@ -1,15 +1,19 @@
...
         required: false
-  push:
-    branches:
-      - support/bot-phosphore
-      - support/bot-phosphore-*
+      currencies:
+        description: currency ids to filter, separated with commas (,)
```
Comment (chabroA) [inline .github/workflows/bot-phosphore.yml:10]: Similar comment as made on "mooncake"

File: .github/workflows/bot-phosphore.yml [hunk 8]
```diff
@@ -1,15 +1,19 @@
 name: "[Bot] Testing with 'Phosphore'"
-run-name: Test with 🤖 Phosphore by @${{github.actor}} with ${{inputs.family}}
+run-name: Test with 🤖 Phosphore by @${{github.actor}} ${{inputs.families &&  format('family={0}',inputs.families) || ''}} ${{inputs.currencies && format('currency={0}',inputs.currencies) || ''}}
```
Comment (chabroA) [inline .github/workflows/bot-phosphore.yml:2]: Shouldn't it be?:
```suggestion
run-name: Test with 🤖 Phosphore by @${{github.actor}} ${{inputs.families &&  format('families={0}',inputs.families) || ''}} ${{inputs.currencies && format('currencies={0}',inputs.currencies) || ''}}
```

File: .github/workflows/bot-oxygen.yml [hunk 9]
```diff
@@ -1,15 +1,19 @@
 name: "[Bot] Testing with 'Oxygen'"
-run-name: Test with 🤖 Oxygen by @${{github.actor}} with ${{inputs.family}}
+run-name: Test with 🤖 Oxygen by @${{github.actor}} ${{inputs.families &&  format('family={0}',inputs.families) || ''}} ${{inputs.currencies && format('currency={0}',inputs.currencies) || ''}}
```
Comment (chabroA) [inline .github/workflows/bot-oxygen.yml:2]: Shouldn't it be?:

```suggestion
run-name: Test with 🤖 Oxygen by @${{github.actor}} ${{inputs.families &&  format('families={0}',inputs.families) || ''}} ${{inputs.currencies && format('currencies={0}',inputs.currencies) || ''}}
```

File: .github/workflows/bot-silicium.yml [hunk 10]
```diff
@@ -1,15 +1,19 @@
 name: "[Bot] Testing with 'Silicium'"
-run-name: Test with 🤖 Silicium by @${{github.actor}} with ${{inputs.family}}
+run-name: Test with 🤖 Silicium by @${{github.actor}} ${{inputs.families &&  format('family={0}',inputs.families) || ''}} ${{inputs.currencies && format('currency={0}',inputs.currencies) || ''}}
```
Comment (chabroA) [inline .github/workflows/bot-silicium.yml:2]: Shouldn't it be?:
```suggestion
run-name: Test with 🤖 Silicium by @${{github.actor}} ${{inputs.families &&  format('families={0}',inputs.families) || ''}} ${{inputs.currencies && format('currencies={0}',inputs.currencies) || ''}}
```

File: .github/workflows/bot-silicium.yml [hunk 11]
```diff
@@ -1,15 +1,19 @@
...
 on:
   workflow_dispatch:
     inputs:
-      family:
-        description: coin family to filter with (if any)
+      families:
+        description: coin family filter, separated with commas (,)
```
Comment (chabroA) [inline .github/workflows/bot-silicium.yml:7]: Similar comment as made on "mooncake"

File: .github/workflows/bot-silicium.yml [hunk 12]
```diff
@@ -1,15 +1,19 @@
...
         required: false
-  push:
-    branches:
-      - support/bot-silicium
-      - support/bot-silicium-*
+      currencies:
+        description: currency ids to filter, separated with commas (,)
```
Comment (chabroA) [inline .github/workflows/bot-silicium.yml:10]: Similar comment as made on "mooncake"

File: .github/workflows/bot-mere-denis.yml [hunk 13]
```diff
@@ -1,20 +1,19 @@
 name: "[Bot] Testing with 'Mère Denis'"
-run-name: Test with 🤖 Mère Denis by @${{github.actor}} with ${{inputs.family}}
+run-name: Test with 🤖 Mère Denis by @${{github.actor}} ${{inputs.families &&  format('family={0}',inputs.families) || ''}} ${{inputs.currencies && format('currency={0}',inputs.currencies) || ''}}
```
Comment (chabroA) [inline .github/workflows/bot-mere-denis.yml:2]: Shouldn't it be?:
```suggestion
run-name: Test with 🤖 Mère Denis by @${{github.actor}} ${{inputs.families &&  format('families={0}',inputs.families) || ''}} ${{inputs.currencies && format('currencies={0}',inputs.currencies) || ''}}
```

File: .github/workflows/bot-mooncake.yml [hunk 14]
```diff
@@ -1,15 +1,19 @@
 name: "[Bot] Testing with 'Mooncake'"
-run-name: Test with 🤖 Mooncake by @${{github.actor}} with ${{inputs.family}}
+run-name: Test with 🤖 Mooncake by @${{github.actor}} ${{inputs.families &&  format('family={0}',inputs.families) || ''}} ${{inputs.currencies && format('currency={0}',inputs.currencies) || ''}}
```
Comment (chabroA) [inline .github/workflows/bot-mooncake.yml:2]: Shouldn't it be?:
```suggestion
run-name: Test with 🤖 Mooncake by @${{github.actor}} ${{inputs.families &&  format('families={0}',inputs.families) || ''}} ${{inputs.currencies && format('currencies={0}',inputs.currencies) || ''}}
```

File: .github/workflows/bot-mere-denis.yml [hunk 15]
```diff
@@ -1,20 +1,19 @@
...
 on:
   workflow_dispatch:
     inputs:
-      family:
-        description: coin family to filter with (if any)
+      families:
+        description: coin family filter, separated with commas (,)
```
Comment (chabroA) [inline .github/workflows/bot-mere-denis.yml:7]: Similar comment as made on "mooncake"

File: .github/workflows/bot-mere-denis.yml [hunk 16]
```diff
@@ -1,20 +1,19 @@
...
+        required: false
+      currencies:
+        description: currency ids to filter, separated with commas (,)
```
Comment (chabroA) [inline .github/workflows/bot-mere-denis.yml:10]: Similar comment as made on "mooncake"

File: .github/workflows/bot-nitrogen.yml [hunk 17]
```diff
@@ -1,20 +1,19 @@
 name: "[Bot] Testing with 'Nitrogen'"
-run-name: Test with 🤖 Nitrogen by @${{github.actor}} with ${{inputs.family}}
+run-name: Test with 🤖 Nitrogen by @${{github.actor}} ${{inputs.families &&  format('family={0}',inputs.families) || ''}} ${{inputs.currencies && format('currency={0}',inputs.currencies) || ''}}
 on:
   workflow_dispatch:
     inputs:
-      family:
-        description: coin family to filter with (if any)
+      families:
+        description: coin family filter, separated with commas (,)
```
Comment (chabroA) [inline .github/workflows/bot-nitrogen.yml:7]: Similar comment as made on "mooncake"

File: .github/workflows/bot-nitrogen.yml [hunk 18]
```diff
@@ -1,20 +1,19 @@
...
+        required: false
+      currencies:
+        description: currency ids to filter, separated with commas (,)
```
Comment (chabroA) [inline .github/workflows/bot-nitrogen.yml:10]: Similar comment as made on "mooncake"

File: .github/workflows/bot-nitrogen.yml [hunk 19]
```diff
@@ -1,20 +1,19 @@
 name: "[Bot] Testing with 'Nitrogen'"
-run-name: Test with 🤖 Nitrogen by @${{github.actor}} with ${{inputs.family}}
+run-name: Test with 🤖 Nitrogen by @${{github.actor}} ${{inputs.families &&  format('family={0}',inputs.families) || ''}} ${{inputs.currencies && format('currency={0}',inputs.currencies) || ''}}
```
Comment (chabroA) [inline .github/workflows/bot-nitrogen.yml:2]:
```suggestion
run-name: Test with 🤖 Nitrogen by @${{github.actor}} ${{inputs.families &&  format('families={0}',inputs.families) || ''}} ${{inputs.currencies && format('currencies={0}',inputs.currencies) || ''}}
```

**Review by chabroA — APPROVED** [review]
Looks good to me overall, some UX improvements proposed

File: .github/workflows/bot-nonreg-nitrogen.yml (hunk 3 above)
Comment (chabroA) [reply .github/workflows/bot-nonreg-nitrogen.yml:36]: I think this is linked to (cf. description of the PR):

> Disabled on non reg : Bitcoin, Ethereum as it costs too much and we test on testnets
>Weekly basis: qtum,decred,cardano,axelar,cosmos,secret_network,avalanche_c_chain,bsc,filecoin,tron,cronos,fantom,boba,telos_evm,polygon_zk_evm,polkadot
>Daily basis : the rest

So basically:
- Bitcoin and Ethereum are always disabled on non reg (weekly or daily)
- On the weekly non reg, only `qtum,decred,cardano,axelar,cosmos,secret_network,avalanche_c_chain,bsc,filecoin,tron,cronos,fantom,boba,telos_evm,polygon_zk_evm,polkadot` run
- On a daily basis, everything else runs (i.e, everything that does not run on a weekly basis, also excluding bitcoin and ethereum that never run on non reg)

Comment (chabroA) [reply .github/workflows/bot-nonreg-nitrogen.yml:36]: Could be worth a comment directly in the workflow files for better understanding (present and future), for example:

For `bot-nonreg-oxygen`:
```
# We only run these currencies on a weekly basis because (...)
BOT_FILTER_CURRENCIES: (...)
```

For `bot-nonreg-nitrogen`:
```
# We don't run these currencies daily because (...)
BOT_DISABLED_CURRENCIES: (...)
```

File: libs/ledger-live-common/src/bot/index.ts [hunk 20]
```diff
@@ -105,39 +99,72 @@ function makeAppJSON(accounts: Account[]) {
...
+    if (disabledFamilies.includes(family)) {
+      // We don't want to test disabled families
       continue;
     }
 
+    const familySpecs = allSpecs[family];
+
     for (const key in familySpecs) {
       let spec: AppSpec<any> = familySpecs[key];
+      if (filteredCurrencies.length > 0 && !filteredCurrencies.includes(spec.currency.id)) {
```
Comment (hzheng-ledger) [inline libs/ledger-live-common/src/bot/index.ts:135]: I think you removed the "findCryptoCurrencyByKeyword" logic from previous code.
The strings in "filteredCurrencies" are not necessary currency id, they can be tick, id, name or keywords.
Maybe you can preprocess filteredCurrencies array with "findCryptoCurrencyByKeyword"

File: tools/actions/composites/bot/action.yml [hunk 21]
```diff
@@ -63,7 +72,10 @@ runs:
         VERBOSE_FILE: botreport/logs.txt
         GITHUB_RUN_ID: ${{ github.run_id }}
         GITHUB_WORKFLOW: ${{ github.workflow }}
-        BOT_FILTER_FAMILY: ${{ inputs.BOT_FILTER_FAMILY }}
+        BOT_FILTER_FAMILIES: ${{ inputs.BOT_FILTER_FAMILIES }}
```
Comment (hzheng-ledger) [inline tools/actions/composites/bot/action.yml:75]: I see you renamed the BOT_FILTER_FAMILY with BOT_FILTER_FAMILIES.
You can check all the BOT_FILTER_FAMILY in our codebase. I found maybe you still forgot to update some yml files.

**Review by hzheng-ledger — CHANGES_REQUESTED** [review]
It is OK in general. 
Just need some minor change. 
Do not forget to update https://github.com/LedgerHQ/ledger-live/wiki/LLC:bot#the-different-bots
Thanks

File: libs/ledger-live-common/src/bot/index.ts (hunk 20 above)
Comment (hedi-edelbloute) [reply libs/ledger-live-common/src/bot/index.ts:135]: Yes,  but I think we should filter by currency id and family names

We know what we want to test

To be discussed

File: tools/actions/composites/bot/action.yml (hunk 21 above)
Comment (hedi-edelbloute) [reply tools/actions/composites/bot/action.yml:75]: Could be the case ! Will check

File: .github/workflows/bot-nitrogen.yml (hunk 2 above)
Comment (valpinkman) [reply .github/workflows/bot-nitrogen.yml:14]: I never said it was mandatory, I was actually asking why it was removed and the answer is enough. You can remove the on push trigger if it is indeed never used 👍

File: .github/workflows/bot-mooncake.yml [hunk 22]
```diff
@@ -1,15 +1,19 @@
...
-        description: coin family to filter with (if any)
+      families:
+        description: coin family filter, separated with commas (,)
         required: false
-  push:
-    branches:
-      - support/bot-mooncake
-      - support/bot-mooncake-*
+      currencies:
+        description: currency ids to filter, separated with commas (,)
```
[the remark this answers is not in the document]
Comment (hedi-edelbloute) [reply .github/workflows/bot-mooncake.yml]: Done

File: .github/workflows/bot-oxygen.yml (hunk 4 above)
Comment (hedi-edelbloute) [reply .github/workflows/bot-oxygen.yml]: Done

File: .github/workflows/bot-oxygen.yml (hunk 5 above)
Comment (hedi-edelbloute) [reply .github/workflows/bot-oxygen.yml]: Done

File: .github/workflows/bot-phosphore.yml (hunk 6 above)
Comment (hedi-edelbloute) [reply .github/workflows/bot-phosphore.yml]: Done

File: .github/workflows/bot-phosphore.yml (hunk 7 above)
Comment (hedi-edelbloute) [reply .github/workflows/bot-phosphore.yml]: Done

File: .github/workflows/bot-silicium.yml (hunk 11 above)
Comment (hedi-edelbloute) [reply .github/workflows/bot-silicium.yml]: Done

File: .github/workflows/bot-silicium.yml (hunk 12 above)
Comment (hedi-edelbloute) [reply .github/workflows/bot-silicium.yml]: Done

File: .github/workflows/bot-mere-denis.yml (hunk 15 above)
Comment (hedi-edelbloute) [reply .github/workflows/bot-mere-denis.yml:7]: Done

File: .github/workflows/bot-mere-denis.yml (hunk 16 above)
Comment (hedi-edelbloute) [reply .github/workflows/bot-mere-denis.yml]: Done

File: .github/workflows/bot-nitrogen.yml (hunk 18 above)
Comment (hedi-edelbloute) [reply .github/workflows/bot-nitrogen.yml]: Done

File: tools/actions/composites/bot/action.yml (hunk 21 above)
Comment (hedi-edelbloute) [reply tools/actions/composites/bot/action.yml:75]: Indeeed, nice catch, done

File: .github/workflows/bot-mooncake.yml [hunk 23]
```diff
@@ -1,15 +1,19 @@
 name: "[Bot] Testing with 'Mooncake'"
-run-name: Test with 🤖 Mooncake by @${{github.actor}} with ${{inputs.family}}
+run-name: Test with 🤖 Mooncake by @${{github.actor}} ${{inputs.families &&  format('family={0}',inputs.families) || ''}} ${{inputs.currencies && format('currency={0}',inputs.currencies) || ''}}
 on:
   workflow_dispatch:
     inputs:
-      family:
-        description: coin family to filter with (if any)
+      families:
+        description: coin family filter, separated with commas (,)
```
[the remark this answers is not in the document]
Comment (hedi-edelbloute) [reply .github/workflows/bot-mooncake.yml]: Done

File: .github/workflows/bot-phosphore.yml (hunk 8 above)
Comment (hedi-edelbloute) [reply .github/workflows/bot-phosphore.yml]: It's visual only but ok,done

File: .github/workflows/bot-oxygen.yml (hunk 9 above)
Comment (hedi-edelbloute) [reply .github/workflows/bot-oxygen.yml]: Done

File: .github/workflows/bot-silicium.yml (hunk 10 above)
Comment (hedi-edelbloute) [reply .github/workflows/bot-silicium.yml]: Done

File: .github/workflows/bot-mere-denis.yml (hunk 13 above)
Comment (hedi-edelbloute) [reply .github/workflows/bot-mere-denis.yml]: Done

File: .github/workflows/bot-mooncake.yml (hunk 14 above)
Comment (hedi-edelbloute) [reply .github/workflows/bot-mooncake.yml]: Done

File: .github/workflows/bot-nitrogen.yml (hunk 17 above)
Comment (hedi-edelbloute) [reply .github/workflows/bot-nitrogen.yml]: Done

File: .github/workflows/bot-nitrogen.yml (hunk 19 above)
Comment (hedi-edelbloute) [reply .github/workflows/bot-nitrogen.yml]: Done

**Review by hzheng-ledger — APPROVED** [review]
Looks good to me after some offline discussion. Good job
